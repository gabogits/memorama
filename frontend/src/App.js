import React, { useEffect, useReducer } from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import GameContext from "./context/GameContext";
import { GameReducer } from "./context/GameReducer";
import { PrivateRoute } from "./routers/PrivateRoute";
import { PublicRoute } from "./routers/PublicRoute";
import { GameRoutes } from "./routers/GameRoutes";
import Player from "./pages/Player";
import Ranking from "./pages/Ranking";

const init = () => {
  let players = localStorage.getItem("dataPlayers");
  players = players ? JSON.parse(players) : [];
  return {
    numberPlayers: players ? players.length : 0,
    dataPlayers: players ? players : [],
    arrayImages: [],
    category: null,
  };
};
const App = () => {
  const [state, dispatch] = useReducer(GameReducer, {}, init);
  const { dataPlayers } = state;

  useEffect(() => {
    localStorage.setItem("dataPlayers", JSON.stringify(dataPlayers));
  }, [dataPlayers]);
  return (
    <GameContext.Provider value={{ state, dispatch }}>
      <Router>
        <Switch>
          <Route exact path="/ranking" component={Ranking} />
          <PublicRoute
            exact
            path="/"
            component={Player}
            dataPlayers={dataPlayers.length}
          />
          <PrivateRoute
            path="/"
            component={GameRoutes}
            dataPlayers={dataPlayers.length}
          />
        </Switch>
      </Router>
    </GameContext.Provider>
  );
};

export default App;
