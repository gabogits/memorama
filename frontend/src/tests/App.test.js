import { mount } from "enzyme";
import App from "../App";

describe("Pruebas en <App />", () => {
  Storage.prototype.setItem = jest.fn();
  Storage.prototype.getItem = jest.fn();
  test("Debe de mostar la routa / al inicio de la aplicacion", () => {
    const wrapper = mount(<App />);
    expect(wrapper).toMatchSnapshot();
    expect(localStorage.setItem).toHaveBeenCalledWith("dataPlayers", "[]");

    expect(wrapper.find("label").text().trim()).toBe(
      "Elige el n√∫mero de Jugadores"
    );
  });
});
