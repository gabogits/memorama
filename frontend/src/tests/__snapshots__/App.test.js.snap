// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <App /> Debe de mostar la routa / al inicio de la aplicacion 1`] = `
<main>
  <header>
    <div
      className=" container container-header"
    >
      <div
        className="logo"
      >
        <figure>
          <a
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo-desk"
            >
              MEMORAMA
            </span>
            <span
              className="logo-mobile"
            >
              M
            </span>
             
          </a>
        </figure>
      </div>
      <nav>
        <div
          className="nav-left"
        />
        <div
          className="nav-right"
        >
          <div
            className="nav-list"
          >
            <ul>
              <li>
                <button
                  onClick={[Function]}
                >
                  Inicio
                </button>
              </li>
              <li>
                <a
                  aria-current={null}
                  href="/ranking"
                  onClick={[Function]}
                >
                  Ranking
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
  </header>
  <section
    className="form top-bottom-space"
  >
    <div
      className="container"
    >
      <section
        className="section-format col"
      >
        <div
          className="field"
        >
          <label>
            Elige el n√∫mero de Jugadores 
          </label>
          <div
            className="buttons-row"
          >
            <button
              className="btn-primary   btn-size-2  btn-color-1 "
              key="1"
              name="numberPlayers"
              onClick={[Function]}
            >
              Un jugador
            </button>
            <button
              className="btn-primary   btn-size-2  btn-color-1 "
              key="2"
              name="numberPlayers"
              onClick={[Function]}
            >
              Dos jugadores
            </button>
          </div>
        </div>
      </section>
    </div>
  </section>
</main>
`;
